app/controllers/reviews_controller.rb

```diff
@@ -0,0 +1,39 @@
+class ReviewsController < ApplicationController
+  before_action :set_user
+
+  def index
+    if params[:query].present?
+      author = User.where("name ILIKE ?", "%#{params[:query]}%").first
+      @reviews = Review.where(receiver: @user, author: author)
+    else
+      @reviews = Review.where(receiver: @user)
+    end
+  end
+
+  def new
+    @review = Review.new
+  end
+
+  def create
+    @review = Review.new(review_params)
+    review.author = current_user
+    if @review.save
+       redirect_to user_reviews_path(@user)
+    else
+      render :new
+    end
+  end
+
+  private
+
+  def review_params
+    params.require(:review).permit(:content, :rating, :barber_review)
```

Comment:
We cannot ask the user to specify if it is a :barber_review, we have to figure it out on our own, based on which dashboard (client or barber) the new review button is clicked from. So we need not whitelist :barber_review